### PR TITLE
Revise helm / AMI dependency in OSS suite release

### PR DESCRIPTION
**Pull request by izgeri**
- Remove helm / AMI repos from suite, since they are mechanisms for distributing product releases and are not products themselves
- Add items to high level technical design on updating the helm chart / re-publishing an AMI (TBD if Conjur core does this automatically already)
- Add K8s secrets provider to the suite

### CODE REVIEW

File: design/oss_suite_release.md [hunk 1]
```diff
@@ -230,14 +229,16 @@ will include:
     links to artifacts for each component version included in the release
   - Note: when possible, links to artifacts should link directly to the canonical
     source, rather than linking to the component README (eg link directly to Dockerhub, etc)
+  - Bumping the version of Conjur pinned in the [Conjur helm chart](https://github.com/cyberark/conjur-oss-helm-chart)
```
Comment (alexkalish) [inline design/oss_suite_release.md]: I thought we decided this was out of scope.  Why the change to bring it in?

File: design/oss_suite_release.md [hunk 2]
```diff
@@ -76,6 +74,7 @@ In addition, as part of building this release, we'll also be creating:
   - [Jenkins](https://github.com/cyberark/conjur-credentials-plugin)
   - Kubernetes / OpenShift
     - [Kubernetes authenticator client](https://github.com/cyberark/conjur-authn-k8s-client)
+    - [Conjur Kubernetes Secrets Provider](https://github.com/cyberark/secrets-provider-for-k8s)
```
Comment (alexkalish) [inline design/oss_suite_release.md]: Probably worth a discussion with IL, just so that understand this will be in the suite and what the repo requirements will be.

File: design/oss_suite_release.md (hunk 1 above)
Comment (izgeri) [reply design/oss_suite_release.md]: we don't list this as out of scope in the doc, and I don't expect much effort needed to submit a PR to the helm chart repo with a conjur version bump. I think this is doable, and I'm fine including it based on my current understanding of it
